tracee-ebpf/btfinfo.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"bufio"
+	"errors"
+	"os"
+	"strings"
+	"syscall"
+)
+
+// Helpers
+
+func FileExists(path string) bool {
+	_, err := os.Stat(path)
+	return err == nil
+}
+
+func FileGrepUnique(path string, needle string) string {
+	file, _ := os.Open(path)
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	for i := 1; scanner.Scan(); i++ {
+		if strings.Contains(scanner.Text(), needle) {
+			return scanner.Text()
+		}
+	}
+
+	return ""
+}
+
+func ByteToString(all [65]int8) string {
+	var temp [65]byte
+	for i, b := range all {
+		temp[i] = byte(b)
+	}
+	conv := string(temp[:])
+	conv = conv[:strings.Index(conv, "\x00")]
+
+	return conv
+}
+
+func KernelVersion() string {
+	utsname := syscall.Utsname{}
+	err := syscall.Uname(&utsname)
+	if err != nil {
+		return ""
+	}
+
+	return ByteToString(utsname.Release)
+}
+
+// BTFinfo
+
+type BTFinfo struct {
+	distroid     string
+	distrover    string
+	distrokernel string
+}
+
+func (btfi *BTFinfo) GetDistroId() (string, error) {
+	if btfi.distroid == "" {
+		return "", errors.New("should call discover first")
+	}
+	return btfi.distroid, nil
+}
+
+func (btfi *BTFinfo) GetDistroVer() (string, error) {
+	if btfi.distrover == "" {
+		return "", errors.New("should call discover first")
+	}
+	return btfi.distrover, nil
+}
+
+func (btfi *BTFinfo) GetDistroKernel() (string, error) {
+	if btfi.distrokernel == "" {
+		return "", errors.New("should call discover first")
+	}
+	return btfi.distrokernel, nil
+}
+
+func (btfi *BTFinfo) DiscoverDistroByKernel() error {
+	// tries to discover running distribution by kernel version/name
```

Comment:
You can put this in a GoDoc comment

Reply:
oh, ok, let me read about it and will do.